### PR TITLE
updated Cohort api

### PR DESCRIPTION
**Pull request by mayankpatel20**
createCohort, findCohortById, findAllCohort, DeleteCohortById methods added.

### CODE REVIEW

**Comment by edchancdn** [comment]
The package paths does not seem match with the base code.

Pls. create feature branch from develop branch.
- git checkout origin develop
- git pull
- git checkout -b feature/ASP-5
- git rebase develop

For the package structure, let's follow the convention from our last project (library-management):
entity - package contains all entities.
repo - package contains repository interfaces with suffix of "repo" or "repository".
rest - contains the rest controllers with suffix of "api".
controller - contains the controllers for the UI.

File: src/main/java/co/pragra/attendancesystem/controller/CohortController.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+public class CohortController {
+
+    @Autowired
+    private CohortService service;
+
```
Comment (edchancdn) [inline src/main/java/co/pragra/attendancesystem/controller/CohortController.java]: pls. add a Put mapping for updateCohort

File: src/main/java/co/pragra/attendancesystem/service/CohortService.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
+package co.pragra.attendancesystem.service;
+
```
Comment (edchancdn) [inline src/main/java/co/pragra/attendancesystem/service/CohortService.java]: This file is not needed anymore.

We are now using interface that extends JpaRepository. 

@Repository
public interface CohortRepo extends JpaRepository<Cohort, Long> {
}

**Review by edchancdn — CHANGES_REQUESTED** [review]
Pls. see comments

**Comment by Hac90111** [comment]
@edchancdn 
I believe,  there should be only one package for apis, a controller, we will work on Ui later.  No need for two separate  rest and controller package

File: src/main/java/co/pragra/attendancesystem/controller/CohortController.java (hunk 1 above)
Comment (mayankpatel20) [reply src/main/java/co/pragra/attendancesystem/controller/CohortController.java]: working on update method and test methods

File: src/main/java/co/pragra/attendancesystem/service/CohortService.java (hunk 2 above)
Comment (mayankpatel20) [reply src/main/java/co/pragra/attendancesystem/service/CohortService.java]: Ok will remove the service layer.

**Comment by mayankpatel20** [comment]
> The package paths does not seem match with the base code.
> 
> Pls. create feature branch from develop branch.
> 
> * git checkout develop
> * git pull
> * git checkout -b feature/ASP-5
> 
> For the package structure, let's follow the convention from our last project (library-management): entity - package contains all entities. repo - package contains repository interfaces with suffix of "repo" or "repository". rest - contains the rest controllers with suffix of "api". controller - contains the controllers for the UI.

Hi Edward, for your comment above, I htink we cna only have the Controller layer. and we do not need the rest package

File: src/main/resources/application.yml [hunk 3]
```diff
@@ -9,5 +9,10 @@ spring:
       enabled: true
   jpa:
     hibernate:
-      ddl-auto: validate
+      ddl-auto: create-drop
```
Comment (edchancdn) [inline src/main/resources/application.yml]: Minor: we can leave ddl-auto as validate

File: src/main/java/co/pragra/attendancesystem/controller/CohortController.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+import co.pragra.attendancesystem.repo.CohortRepo;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.Optional;
+
+@Controller
```
Comment (edchancdn) [inline src/main/java/co/pragra/attendancesystem/controller/CohortController.java]: For an API this should be @RestController

File: src/main/java/co/pragra/attendancesystem/controller/CohortController.java [hunk 5]
```diff
@@ -0,0 +1,75 @@
+package co.pragra.attendancesystem.controller;
```
Comment (edchancdn) [inline src/main/java/co/pragra/attendancesystem/controller/CohortController.java]: Pls. refactor to rename this to CohortApi.java and move to co.pragra.attendancesystem.rest

CohortController.java will be used by the Cohort UI build.

File: src/test/java/co/pragra/attendancesystem/controller/CohortControllerTest.java [hunk 6]
```diff
@@ -0,0 +1,91 @@
+package co.pragra.attendancesystem.controller;
```
Comment (edchancdn) [inline src/test/java/co/pragra/attendancesystem/controller/CohortControllerTest.java]: Pls. refactor to be renamed CohortApiTest.java and move to co.pragra.attendancesystem.rest package in src/test/java

**Review by edchancdn — CHANGES_REQUESTED** [review]
Just minor changes.

File: src/main/java/co/pragra/attendancesystem/entity/Cohort.java [hunk 7]
```diff
@@ -26,8 +26,10 @@ public class Cohort {
     private Date startDate;
     @Column(name = "END_DATE", nullable = false)
     private Date endDate;
-    @ManyToMany
+
+    @ManyToMany(cascade = CascadeType.ALL)
```
Comment (edchancdn) [inline src/main/java/co/pragra/attendancesystem/entity/Cohort.java]: For now, let's make is a simple uni-directional relationship.
Let's keep @ManyToMany with no cascade

File: src/main/java/co/pragra/attendancesystem/entity/Cohort.java [hunk 8]
```diff
@@ -26,8 +26,10 @@ public class Cohort {
...
     private List<Student> students;
-    @ManyToMany
+
+    @ManyToMany(cascade = CascadeType.ALL)
```
Comment (edchancdn) [inline src/main/java/co/pragra/attendancesystem/entity/Cohort.java]: For now, let's make is a simple uni-directional relationship.
Let's keep @ManyToMany with no cascade

**Review by edchancdn — DISMISSED** [review]

**Review by edchancdn — APPROVED** [review]

**Pull request merged**